src/fparser/two/Fortran2008.py

```diff
@@ -719,6 +726,42 @@ class Do_Term_Action_Stmt(Do_Term_Action_Stmt_2003):  # R826
     subclass_names = ["Action_Stmt_C816"]
 
 
+class Alloc_Opt(Alloc_Opt_2003):
+    """
+    Fortran2008 rule R627
+    alloc-opt is ERRMSG = errmsg-variable
+              or MOLD = source-expr
+              or SOURCE = source-expr
+              or STAT = stat-variable
+
+    Extends the Fortran2003 version of this class by updating the keyword
+    pairs (used in match) with support for MOLD.
+
+    """
+
+    _keyword_pairs = [
+        ("STAT", Stat_Variable),
+        ("ERRMSG", Errmsg_Variable),
+        ("SOURCE", Source_Expr),
+        ("MOLD", Source_Expr),
```

Comment:
Even if it is the same in practice, should this match Expr directly? Source_Expr is unrelated to MOLD

Reply:
The standard specifies `Source_Expr` so I went with that.